src/batch-transaction.js

```diff
@@ -0,0 +1,343 @@
+    });
+
+    if (resp.transaction) {
+      self.id = resp.transaction.id;
+      self.readTimestamp = resp.transaction.readTimestamp;
+    }
+
+    callback(null, partitions, resp);
+  });
+};
+
+/**
+ * @typedef {object} ReadPartition
+ * @mixes ReadRequestOptions
+ * @property {string} partitionToken The partition token.
+ */
+/**
+ * @typedef {array} CreateReadPartitionsResponse
+ * @property {ReadPartition[]} 0 List of read partitions.
+ * @property {object} 1 The full API response.
+ */
+/**
+ * @callback CreateReadPartitionsCallback
+ * @param {?Error} err Request error, if any.
+ * @param {ReadPartition[]} partitions List of read partitions.
+ * @param {object} apiResponse The full API response.
+ */
+/**
+ * Creates a set of read partitions that can be used to execute a read
+ * operation in parallel. Partitions become invalid when the transaction used
+ * to create them is closed.
+ *
+ * @param {ReadRequestOptions} options Configuration object, describing what to
+ *     read from.
+ * @param {CreateReadPartitionsCallback} [callback] Callback function.
+ * @returns {Promise<CreateReadPartitionsResponse>}
+ *
+ * @example <caption>include:samples/batch.js</caption>
+ * region_tag:create_read_partitions
+ */
+BatchTransaction.prototype.createReadPartitions = function(options, callback) {
+  var reqOpts = codec.encodeRead(options);
+  var gaxOpts = options.gaxOptions;
+
+  if (gaxOpts) {
+    delete reqOpts.gaxOptions;
+  }
+
+  this.createPartitions_(
+    {
+      client: 'SpannerClient',
+      method: 'partitionRead',
+      reqOpts,
+      gaxOpts,
+    },
+    callback
+  );
+};
+
+/**
+ * Executes partition.
+ *
+ * @see {@link Transaction#read} when using {@link ReadPartition}.
+ * @see {@link Transaction#run} when using {@link QueryParition}.
+ *
+ * @param {ReadPartition|QueryParition} partition The partition object.
+ * @param {TransactionRequestReadCallback|RunCallback} callback Callback
+ *     function.
+ * @returns {Promise<RunResponse>|Promise<TransactionRequestReadResponse>}
+ *
+ * @example <caption>include:samples/batch.js</caption>
+ * region_tag:execute_partition
+ */
+BatchTransaction.prototype.execute = function(partition, callback) {
+  if (is.string(partition.table)) {
+    this.read(partition.table, partition, callback);
+    return;
+  }
+
+  this.run(partition, callback);
+};
+
+/**
+ * Executes partition in streaming mode.
+ *
+ * @see {@link Transaction#createReadStream} when using {@link ReadPartition}.
+ * @see {@link Transaction#runStream} when using {@link QueryParition}.
+ *
+ * @param {ReadPartition|QueryPartition} partition The partition object.
+ * @returns {ReadableStream} A readable stream that emits rows.
+ *
+ * @example
+ * const Spanner = require('@google-cloud/spanner');
+ * const spanner = new Spanner();
+ *
+ * const instance = spanner.instance('my-instance');
+ * const database = instance.database('my-database');
+ * const transaction = database.batchTransaction(identifier);
```

Comment:
Is `identifier` supposed to be defined?

Reply:
No, the idea behind this API is that you create a single transaction to be shared amongst several workers. A user can create an identifier via `transaction.identifier()`. How they pass the identifier to other workers is an implementation detail, so I chose to omit it.

Reply:
I know we don't isolate examples and test them anymore, so runnable JS isn't exactly important. Still, is it possible to save users from going "wait, where is `identifier` defined?" and just put something fake in? If it's too complicated, I suppose even:

```js
const identifier = {};
const transaction = database.batchTransaction(identifier);
```

Reply:
I think stubbing the identifier is more confusing, so I just rewrote a section of the example to avoid using it entirely.